### PR TITLE
Additional install script option to install nightly/release builds

### PR DESCRIPTION
**Pull request by carlos-logro**
issue: #219 

# Changes
Asks user which version he would like installed and from there do the necessary installation process.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: webhooks-extension/test/install_dashboard_and_extension.sh [hunk 1]
```diff
@@ -16,7 +16,37 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
 source ${test_dir}/config.sh
 source ${test_dir}/util.sh
 
-echo "Installing Dashoard and webhooks extension into $DASHBOARD_INSTALL_NS namespace"
-install_webhooks_extension $DASHBOARD_INSTALL_NS
-install_dashboard $DASHBOARD_INSTALL_NS
+# Ask user what verison he will like to install
```
Comment (a-roberts) [inline webhooks-extension/test/install_dashboard_and_extension.sh]: Unnecessary comment given the below echo

File: webhooks-extension/test/util.sh [hunk 2]
```diff
@@ -126,69 +126,105 @@ function install_tekton() {
...
   else
-    echo "Dashboard source not detected in $dashboard_dir"
-    read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
+    # Check if user has dashboard source cloned
+    dashboard_dir=$GOPATH/src/github.com/tektoncd/dashboard
+    if [ -d $dashboard_dir ]; then
+      # Notify and depending on [version] we will ko apply or kubectl apply gcr yaml
+      echo -e "\n\nDashboard source detected at $dashboard_dir"
+    else
+      # If not, notify and ask if he wants to clone it
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: Don't need this comment

File: webhooks-extension/test/util.sh [hunk 3]
```diff
@@ -126,69 +126,105 @@ function install_tekton() {
...
+      echo -e "\n\nDashboard source not detected in $dashboard_dir"
+      if [[ $version != "3" ]]; then
+        read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
+      fi
+      if [[ $cloneAndApply =~ ^[Yy]$ ]] || [[ $version == "3" ]]; then
+        # Dashboard source cloned
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: Don't need

File: webhooks-extension/test/util.sh [hunk 4]
```diff
@@ -126,69 +126,105 @@ function install_tekton() {
...
+        curl -L https://github.com/tektoncd/dashboard/releases/download/v0/gcr-tekton-dashboard.yaml | kubectl apply -f -
+      fi
+    fi
+
+    # At this point has the dashboard cloned and the only thing left to do is
+    # ko apply or kubectl apply gcr yaml
+    if [[ $version == "3" ]]; then
+      pushd $GOPATH/src/github.com/tektoncd/dashboard
+      docker login
+      npm install
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: Should be `npm ci`? You use it later so wondering if npm install is intentional here

Comment (carlos-logro) [reply webhooks-extension/test/util.sh]: definitely, thanks Adam!

**Comment by a-roberts** [comment]
@carlos-logro would be useful to include some docs for this PR too in order to advocate the script's usage, what do you think?

**Comment by carlos-logro** [comment]
@a-roberts yes it will indeed help user out! ill go ahead and added it

**Comment by carlos-logro** [comment]
@a-roberts Updated the doc! let me know if the wording is on point

File: webhooks-extension/test/install_dashboard_and_extension.sh [hunk 5]
```diff
@@ -16,7 +16,36 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
...
+
+if [[ $VERSION == "1" ]]; then
+  VERSION_TEXT="CURRENT RELEASES"
+elif [[ $VERSION == "2" ]]; then
+  VERSION_TEXT="NIGHTLY BUILDS"
+else
+  VERSION_TEXT="DEVELOPMENT VERSIONS"
+fi
+
+echo
```
Comment (a-roberts) [inline webhooks-extension/test/install_dashboard_and_extension.sh]: What's this doing here?

File: webhooks-extension/test/install_dashboard_and_extension.sh [hunk 6]
```diff
@@ -16,7 +16,36 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
...
+# Check that GOPATH and KO_DOCKER_REPO are set properly
+check GOPATH
+check KO_DOCKER_REPO "export KO_DOCKER_REPO=docker.io/${DOCKERHUB_USERNAME}"
```
Comment (a-roberts) [inline webhooks-extension/test/install_dashboard_and_extension.sh]: Can you explain/comment/refactor this line please? Confusing on reading, could just be missing a newline

File: webhooks-extension/test/install_dashboard_and_extension.sh [hunk 7]
```diff
@@ -16,7 +16,36 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
...
+
+echo -e "\n\nInstalling the $VERSION_TEXT of the Dashoard and webhooks extension into $DASHBOARD_INSTALL_NS namespace.\n"
```
Comment (a-roberts) [inline webhooks-extension/test/install_dashboard_and_extension.sh]: Tekton Dashboard*
Webhooks Extension

File: webhooks-extension/test/util.sh [hunk 8]
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
...
-
-  dashboard_dir=$GOPATH/src/github.com/tektoncd/dashboard 
-  if [ -d $dashboard_dir ]; then 
-    echo "Dashboard source detected at $dashboard_dir"
-    read -p "Do you wish to ko apply the current source? " -n 1 -r apply
+
+  version=$2
+  # Current Release
+  if [[ $version == "1" ]]; then
+    kubectl apply --filename https://github.com/tektoncd/dashboard/releases/download/v0.2.0/release.yaml
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: Is there a *latest release* URL we could use instead so we needn't keep updating this when we do cut a new one?

File: webhooks-extension/test/util.sh [hunk 9]
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
...
   else
-    echo "Dashboard source not detected in $dashboard_dir"
-    read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
+    # Check if user has dashboard source cloned
+    dashboard_dir=$GOPATH/src/github.com/tektoncd/dashboard
+    if [ -d $dashboard_dir ]; then
+      # Notify and depending on [version] we will ko apply or kubectl apply gcr yaml
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: Notify? Notify what

File: webhooks-extension/test/util.sh [hunk 10]
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
...
+      echo -e "\n\nDashboard source detected at $dashboard_dir"
+    else
+      echo -e "\n\nDashboard source not detected in $dashboard_dir"
+      if [[ $version != "3" ]]; then
+        read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: What does this line do? The flags after it, isn't immediately obvious and I'd rather not go read the manpage for read 😄

File: webhooks-extension/test/util.sh [hunk 11]
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
...
+    kubectl apply -f config/latest/gcr-tekton-webhooks-extension.yaml
+    popd
+  # Development Version
+  else
+    pushd $GOPATH/src/github.com/tektoncd/experimental/webhooks-extension
+    namespace=$1
     docker login
-    npm install
+    npm ci
+    npm rebuild node-sass
```
Comment (a-roberts) [inline webhooks-extension/test/util.sh]: Is this actually needed? Haven't done it in months

File: webhooks-extension/test/install_dashboard_and_extension.sh (hunk 5 above)
Comment (carlos-logro) [reply webhooks-extension/test/install_dashboard_and_extension.sh]: thought I removed this echo....

File: webhooks-extension/test/install_dashboard_and_extension.sh (hunk 6 above)
Comment (carlos-logro) [reply webhooks-extension/test/install_dashboard_and_extension.sh]: what's confusing here? here it is just checking that the user has correctly specified `GOPATH` and `KO_DOCKER_REPO`

Comment (a-roberts) [reply webhooks-extension/test/install_dashboard_and_extension.sh]: Should it not be

check KO_DOCKER_REPO
"export KO_DOCKER_REPO=docker.io/${DOCKERHUB_USERNAME}"

?

File: webhooks-extension/test/util.sh (hunk 8 above)
Comment (carlos-logro) [reply webhooks-extension/test/util.sh]: going to test with: `https://github.com/tektoncd/dashboard/releases/latest`

File: webhooks-extension/test/util.sh (hunk 9 above)
Comment (carlos-logro) [reply webhooks-extension/test/util.sh]: notifiy the user that they in fact have the dashboard cloned.

File: webhooks-extension/test/util.sh (hunk 10 above)
Comment (carlos-logro) [reply webhooks-extension/test/util.sh]: `-n` flag denotes how many characters to read from the user input
`-r` where the value of the input is going to be stored

File: webhooks-extension/test/util.sh (hunk 11 above)
Comment (carlos-logro) [reply webhooks-extension/test/util.sh]: great point. will remove and see if everything works properly!

File: webhooks-extension/test/install_dashboard_and_extension.sh (hunk 6 above)
Comment (carlos-logro) [reply webhooks-extension/test/install_dashboard_and_extension.sh]: i see the confusion now. so the `check` method takes as arguments the value you want to check and optionally, an example of what that value looks like.

File: webhooks-extension/test/util.sh (hunk 11 above)
Comment (AlanGreene) [reply webhooks-extension/test/util.sh]: `npm rebuild` should only be needed (occasionally) when switching node versions

**Comment by carlos-logro** [comment]
@a-roberts changes made!

File: webhooks-extension/test/install_dashboard_and_extension.sh [hunk 12]
```diff
@@ -16,7 +17,35 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
 source ${test_dir}/config.sh
 source ${test_dir}/util.sh
 
-echo "Installing Dashoard and webhooks extension into $DASHBOARD_INSTALL_NS namespace"
-install_webhooks_extension $DASHBOARD_INSTALL_NS
-install_dashboard $DASHBOARD_INSTALL_NS
+echo -e "\n\nPlease choose which versions you will like to install:"
```
Comment (a-roberts) [inline webhooks-extension/test/install_dashboard_and_extension.sh]: you would* like to install

**Comment by carlos-logro** [comment]
@a-roberts changes made and file moved to the `scripts` folder

File: webhooks-extension/README.md [hunk 13]
```diff
@@ -16,7 +16,10 @@ See our [Getting Started](https://github.com/tektoncd/experimental/blob/master/w
 ### Install Webhook Extension
 
 [Installing Official Release (stable)](./docs/InstallReleaseBuild.md)  
-[Installing Development Build (nightly)](./docs/InstallNightlyBuild.md)  
+[Installing Development Build (nightly)](./docs/InstallNightlyBuild.md)
+
+As a convenience, the **/test/install_dashboard_and_extension.sh** script can be
```
Comment (a-roberts) [inline webhooks-extension/README.md]: scripts/ not test

**Comment by carlos-logro** [comment]
🤦‍♂️🤦‍♂️🤦‍♂️ on it

**Comment by carlos-logro** [comment]
@a-roberts updated!

**Comment by mnuttall** [comment]
Please rebase and we'll get this in asap. Thank you!

**Comment by carlos-logro** [comment]
@mnuttall tested and all choices seem to work except 2 and, based on what Vince and Brandon told me, it seems to be a syntax error in the nightly build extension yaml. Other than that everything is working. Also, should we keep the script under `scripts`, I noticed that the script doesn't work if `test/install_prereqs` has not been ran yet.

**Comment by carlos-logro** [comment]
Updated the script. Reason behind the error was because of an indentation error on the webhhooks extension nightly build yaml. I also moved the script to the test folder primary because for this script to run successfully the prerequisites should already be installed.

**Pull request merged**